rs_client.py

```diff
@@ -1019,16 +1019,17 @@ def _rs_struct(typeobj):
 
         _r('pub type %s%s = base::StructPtr<%s%s>;', typeobj.rs_type, lifetime1,
                 lifetime2, typeobj.ffi_type)
-        typeobj.struct_ptr = True
 
 
 def _rs_accessors(typeobj):
 
     lifetime = "<'a>" if typeobj.has_lifetime else ""
+    # unions always get an impl lifetime to be sure
+    lifetime2 = "<'a>" if typeobj.is_union else lifetime
 
```

Comment:
I prefer to have the lifetime decl in the accessor. Main reason is that some unions (most?) do not have any StructPtr fields. Second, I find it cleaner (which is very personal, I admit).